### DISCUSSION TITLE
Usage with Tanstack router?

### DISCUSSION THREAD

**Discussion by InsaneNaman** [honojs/hono#2606] [General]
Tanstack is coming with Tanstack start (meta framework using vinxi which is based on nitro) but I am looking for ideas/directions to use Tanstack router with hono. Any good example I can refer to?

**Comment by ghost** [comment]
That may not be impossible, but I think it is a bad idea. Like trying to fly an airplane on one AAA battery.

The default router for TanStack Router is in [this file](https://github.com/TanStack/router/blob/main/packages/react-router/src/router.ts) (44.8 kB).
This is roughly seven times the amount of code as [Hono's reg-exp-router](https://github.com/honojs/hono/blob/main/src/router/reg-exp-router/router.ts) (6.2 kB).
This is because it contains essential functions to support [these rich features](https://github.com/TanStack/router/tree/main?tab=readme-ov-file#quick-features).
Rewriting them to use Hono's primitive Router is difficult.
I don't know what features you want in Hono that are not in the Tanstack Router, but I think it would be a good idea to discuss these features with the TanStack community first.

> Also try [HonoX](https://github.com/honojs/honox) made in Hono. (Similar to TanStack Start, lightweight and React-independent)

**Reply by airtonix** [reply]
And what about just treating tanstack router as a client side thing only? 

The reason you're getting all these kinds of requests is simply that HonoX just can't support all the things we want to use with react.

**Comment by airtonix** [comment]
@InsaneNaman You can try this template from cloudflare: https://developers.cloudflare.com/workers/framework-guides/web-apps/more-web-frameworks/hono/

I was able to get tanstack router working with hono this way.

**Comment by bskimball** [comment]
This is how I do it
https://github.com/bskimball/tanstack-hono

**Reply by lalatina11** [reply]
wow that is insane tanstack start with hono integrating, i hope the tanstack developer look at your project dude!

**Reply by xkrfer** [reply]
it’s cool!
